### PR TITLE
refactor(Avatar): Make avatar more versatile

### PR DESCRIPTION
**Pull request by dekkofilms**
Avatar can now take in a size property, and
refactored the styles to not be absolute positioned.
This will help with it being placed in other components.
Cards Primary component also was affected by this change,
and was fixed accordingly.

Link to preview app: https://docs-tvbesfawbx.now.sh/

### CODE REVIEW

File: pages/cards.js [hunk 1]
```diff
@@ -72,21 +64,24 @@ const CardsPage = ({ className }) => (
...
-        </Primary>
-        <SupportingText>Praesent commodo cursus magna, vel scelerisque nisl consectetur et. Integer posuere erat a ante venenatis dapibus posuere velit aliquet. Donec id elit non mi porta gravida at eget metus. Maecenas sed diam eget risus varius blandit sit amet non magna.</SupportingText>
+        <Primary
+          title="Title"
+          subheader="Subhead"
+        />
+        <SupportingText>Praesent commodo cursus magna, vel scelerisque nisl consectetur et.
+        Integer posuere erat a ante venenatis dapibus posuere velit aliquet.
+        Donec id elit non mi porta gravida at eget metus. Maecenas sed diam eget
+        risus varius blandit sit amet non magna.</SupportingText>
```
Comment (martahj) [inline pages/cards.js]: NIT: multiline elements should have the children on a different line from the tags:

```js
<SupportingText>
  Praesent commodo cursus magna, vel scelerisque nisl consectetur et.
</SupportingText>
```

File: src/components/Avatar.js [hunk 2]
```diff
@@ -2,23 +2,27 @@ import React from 'react';
 import styled from 'styled-components';
 
 const AvatarComponent = props => (
-  <div className={`${props.className} smc-avatar`}>
+  <div className={`${props.className} smc-avatar`} size={props.size}>
```
Comment (martahj) [inline src/components/Avatar.js]: The styled component wraps the AvatarComponent, so you don't need to pass `props.size` in order to be able to access `props.size` in the `styled` call.

File: src/components/List/ListItem.js [hunk 3]
```diff
@@ -2,26 +2,28 @@ import React from 'react';
 import styled from 'styled-components';
 import { Avatar } from '../Avatar';
 
-const ListItemComponent = ({ className, children, leftAvatar }) => (
+const ListItemComponent = ({ className, children, leftAvatar, size }) => (
   <li className={className}>
     {leftAvatar && (
-      <Avatar src={leftAvatar} />
+      <div className="smc-list-avatar">
+        <Avatar src={leftAvatar} size={size} />
```
Comment (martahj) [inline src/components/List/ListItem.js]: Since `size` is not used anywhere else in this component, it might make more sense to leave this out and let the user modify the height and width of the avatar div using `styled()`.

If you leave the prop in, you should rename it to something like `avatarSize` to help the user know what component they are setting the size on.

File: pages/cards.js (hunk 1 above)
Comment (dekkofilms) [reply pages/cards.js]: Good catch ... I think I need to configure something in VSCode cause prettier seems to change my formatting to the file whenever I copy and paste something around.

File: src/components/Avatar.js (hunk 2 above)
Comment (dekkofilms) [reply src/components/Avatar.js]: Ah! Good to know!

**Comment by brad-decker** [comment]
The avatars on the list page are not stretched to fill the circle that contains them. You have gray bars on top and bottom

**Comment by dekkofilms** [comment]
@brad-decker eek! fixed that ... although I don't think that was the greatest picture for an example. I know `object-fit` doesn't have IE or Edge support. I can definitely think through another solution if that isn't appealing.

@martahj fixed based on your comments.

File: src/components/Card/Primary.js [hunk 4]
```diff
@@ -1,13 +1,59 @@
...
-import styled from 'styled-components';
+import styled, { css } from 'styled-components';
+import { Avatar } from '../Avatar';
+import typography from '../../mixins/typography';
 
-const PrimaryComponent = props => (
-  <section className={`${props.className} card-primary`}>{props.children}</section>
+const Title = styled.h1`
+  ${typography('body2')}
+  margin: -.063rem 0;
```
Comment (AriLFrankel) [inline src/components/Card/Primary.js]: Where does this number come from?

File: src/components/Card/Primary.js [hunk 5]
```diff
@@ -1,13 +1,59 @@
...
+
+  ${props => props.large && css`
+    ${typography('headline')}
+    margin: 0;
+  `}
+
+  .primary & {
```
Comment (AriLFrankel) [inline src/components/Card/Primary.js]: I don't think this is taking effect.
 
There are other areas in the Card code where similar selectors are used and don't seem to take effect either. It might be a separate PR to remove those.

**Comment by dekkofilms** [comment]
@brad-decker update link: https://docs-uycvryndav.now.sh/

File: src/components/Card/Primary.js [hunk 6]
```diff
@@ -1,13 +1,59 @@
...
+    ${props => props.large && css `
+      padding-top: 8px;
+    `}
+  }
+`;
+
+const Subheader = styled.h2`
+  ${typography('body1')}
+  margin: -.063rem 0;
```
Comment (AriLFrankel) [inline src/components/Card/Primary.js]: same q as above

File: src/components/Card/Primary.js (hunk 4 above)
Comment (dekkofilms) [reply src/components/Card/Primary.js]: Yeah, I am not too sure currently ... I can look into it though. I copy and pasted that over from the Title component that isn't needed anymore if this PR goes through.

File: src/components/Card/Primary.js (hunk 5 above)
Comment (dekkofilms) [reply src/components/Card/Primary.js]: Yeah, good point. I don't mind fixing that stuff up in this PR ... but I was primarily just focused on fixing the styles for the changed Avatar to still work with the Cards.

Comment (brad-decker) [reply src/components/Card/Primary.js]: No need for this pr.

File: src/components/Card/Primary.js (hunk 6 above)
Comment (brad-decker) [reply src/components/Card/Primary.js]: @AriLFrankel the pr actually didn't touch this stuff, @dekkofilms just messed with how it was composed. If we need to tweak the card component we can do so in another PR

**Comment by brad-decker** [comment]
@dekkofilms rebase, and LGTM
